contracts/mainnet/connectors/liquity/main.sol

```diff
@@ -0,0 +1,347 @@
+        _eventName = "LogClaimCollateralFromRedemption(address)";
+        _eventParam = abi.encode(msg.sender);
+    }
+    /* End: Trove */
+
+    /* Begin: Stability Pool */
+
+    /**
+     * @dev Deposit LUSD into Stability Pool
+     * @notice Deposit LUSD into Stability Pool
+     * @param amount Amount of LUSD to deposit into Stability Pool
+     * @param frontendTag Address of the frontend to make this deposit against (determines the kickback rate of rewards)
+     * @param getId Optional storage slot to retrieve the LUSD from
+    */
+    function stabilityDeposit(
+        uint amount,
+        address frontendTag,
+        uint getId
+    ) external returns (string memory _eventName, bytes memory _eventParam) {
+        amount = getUint(getId, amount);
+
+        stabilityPool.provideToSP(amount, frontendTag);
+        
+        _eventName = "LogStabilityDeposit(address,uint,address,uint)";
+        _eventParam = abi.encode(msg.sender, amount, frontendTag, getId);
+    }
+
+    /**
+     * @dev Withdraw user deposited LUSD from Stability Pool
+     * @notice Withdraw LUSD from Stability Pool
+     * @param amount Amount of LUSD to withdraw from Stability Pool
+     * @param setId Optional storage slot to store the withdrawn LUSD
+    */
+    function stabilityWithdraw(
+        uint amount,
+        uint setId
+    ) external returns (string memory _eventName, bytes memory _eventParam) {
+        stabilityPool.withdrawFromSP(amount);
+        setUint(setId, amount);
+
+        _eventName = "LogStabilityWithdraw(address,uint,uint)";
+        _eventParam = abi.encode(msg.sender, amount, setId);
+    }
+
+    /**
+     * @dev Increase Trove collateral by sending Stability Pool ETH gain to user's Trove
+     * @notice Moves user's ETH gain from the Stability Pool into their Trove
+     * @param upperHint Address of the Trove near the upper bound of where the user's Trove will now sit in the ordered Trove list
+     * @param lowerHint Address of the Trove near the lower bound of where the user's Trove will now sit in the ordered Trove list
+    */
+    function stabilityMoveEthGainToTrove(
+        address upperHint,
+        address lowerHint
+    ) external returns (string memory _eventName, bytes memory _eventParam) {
+        stabilityPool.withdrawETHGainToTrove(upperHint, lowerHint);
+
+        _eventName = "LogStabilityMoveEthGainToTrove(address)";
+        _eventParam = abi.encode(msg.sender);
+    }
+    /* End: Stability Pool */
+
+    /* Begin: Staking */
+
+    /**
+     * @dev Sends LQTY tokens from user to Staking Pool
+     * @notice Stake LQTY in Staking Pool
+     * @param amount Amount of LQTY to stake
+     * @param getId Optional storage slot to retrieve the LQTY from
+    */
+    function stake(
+        uint amount,
+        uint getId
+    ) external returns (string memory _eventName, bytes memory _eventParam) {
+        amount = getUint(getId, amount);
+        staking.stake(amount);
```

Comment:
Same here, in all staking operations it would be nice to have setUint’s for LUSD and ETH gains.

Reply:
https://github.com/liquity/dsa-connectors/pull/1/commits/97db94fe595330743385f26c30e15ab28cc18063

Reply:
https://github.com/liquity/dsa-connectors/pull/1/commits/5be5cb68668172047e29323092c574d4155f80f8